ts/Extensions/Annotations/Popup.ts

```diff
@@ -795,162 +875,381 @@ H.Popup.prototype = {
+                rhsCol = parentDiv.querySelectorAll('.' + PREFIX + 'popup-rhs-col')[0] as HTMLElement,
                 isEdit = listType === 'edit',
+                addFormFields = this.indicators.addFormFields,
                 series = (
                     isEdit ?
                         chart.series : // EDIT mode
                         chart.options.plotOptions // ADD mode
-                ),
-                addFormFields = this.indicators.addFormFields,
-                rhsColWrapper: Element,
+                );
+
+            let rhsColWrapper: HTMLElement,
                 indicatorList: HTMLDOMElement,
-                item: HTMLDOMElement;
+                item: HTMLDOMElement,
+                filteredSeriesArray: [Highcharts.FilteredSeries] = [] as any;
 
-            if (!chart) {
+            if (!chart && series) {
                 return;
             }
 
-            // create wrapper for list
-            indicatorList = createElement(UL, {
-                className: PREFIX + 'indicator-list'
-            }, null as any, lhsCol as any);
+            // Filter and sort the series.
+            if (!isEdit && series && !isArray(series)) {
+                // Apply filters only for the 'add' indicator list.
+                filteredSeriesArray = indicators.filterSeries.call(this, series, filter);
+            } else if (series && isArray(series)) {
+                filteredSeriesArray = indicators.filterSeriesArray.call(this, series);
+            }
 
-            rhsColWrapper = rhsCol
-                .querySelectorAll('.' + PREFIX + 'popup-rhs-col-wrapper')[0];
+            // If the list exists remove it from the DOM
+            // in order to create a new one with different filters.
+            if (lhsCol.children[3]) {
+                lhsCol.children[3].remove();
+            }
 
-            objectEach(series, function (
-                serie: (Series|SeriesTypePlotOptions),
-                value: string
-            ): void {
-                const seriesOptions = serie.options;
+            // Create wrapper for list.
+            indicatorList = createElement(
+                UL,
+                {
+                    className: PREFIX + 'indicator-list'
+                },
+                void 0,
+                lhsCol
+            );
 
-                if (
-                    (serie as any).params ||
-                    seriesOptions && (seriesOptions as any).params
-                ) {
+            rhsColWrapper = rhsCol
+                .querySelectorAll('.' + PREFIX + 'popup-rhs-col-wrapper')[0] as HTMLElement;
 
-                    const indicatorNameType = _self.indicators.getNameType(serie as any, value),
-                        indicatorType = indicatorNameType.type;
+            filteredSeriesArray.forEach(function (seriesSet: Highcharts.FilteredSeries): void {
+                const { indicatorFullName, indicatorType, series } = seriesSet;
```

Comment:
Are we allowed to destructure function arguments? 

```ts
filteredSeriesArray.forEach(function (seriesSet: Highcharts.FilteredSeries): void {
  const { indicatorFullName, indicatorType, series } = seriesSet;
```
->
```ts
filteredSeriesArray.forEach(function ({ indicatorFullName, indicatorType, series }): void {
```

@bre1470

Reply:
Unlikely, unless Babel is doing something about it. TypeScript will not convert it, because we have a ES6 target for Babel.

Reply:
With destructurization as Pawel suggested it works as before, however it compiles it a bit odd.
```JS
filteredSeriesArray.forEach(function (_a) {
   var indicatorFullName = _a.indicatorFullName, indicatorType = _a.indicatorType, series = _a.series;
  ...
```

I think the first approach is more readable and results in clearer code after compiling.